src/parser/jobs/drg/modules/Drift.tsx

```diff
@@ -0,0 +1,158 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {Message, Table, Button, Header, Icon} from 'semantic-ui-react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Downtime from 'parser/core/modules/Downtime'
+import {getDataBy} from 'data'
+import Module, {dependency} from 'parser/core/Module'
+import React, {Fragment} from 'react'
+import {Timeline} from 'parser/core/modules/Timeline'
+
+// Buffer (ms) to forgive insignificant drift, we really only care about GCD drift here
+// and not log inconsistencies / sks issues / misguided weaving
+const DRIFT_BUFFER = 1200
+
+const DRIFT_ABILITIES = [
+	ACTIONS.HIGH_JUMP.id,
+	ACTIONS.GEIRSKOGUL.id,
+]
+
+const COOLDOWN_MS = {
+	[ACTIONS.HIGH_JUMP.id]: ACTIONS.HIGH_JUMP.cooldown * 1000,
+	[ACTIONS.GEIRSKOGUL.id]: ACTIONS.GEIRSKOGUL.cooldown * 1000,
+}
+
+class DriftWindow {
+	actionId: number
+	start: number
+	end: number = 0
+	drift: number = 0
+	abilityRotation: CastEvent[] = []
+
+	constructor(actionId: number, start: number) {
+		this.actionId = actionId
+		this.start = start
+	}
+
+	public addAbility(event: CastEvent) {
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+		if (action) {
+			this.abilityRotation.push(event)
+		}
+	}
+
+	public getLastActionId(): number {
+		return this.abilityRotation.slice(-1)[0].ability.guid
+	}
+}
+
+export default class Drift extends Module {
+	static handle = 'drift'
+	static title = t('drg.drift.title')`Ability Drift`
+
+	@dependency private downtime!: Downtime
+	@dependency private timeline!: Timeline
+
+	private driftedWindows: DriftWindow[] = []
+
+	private currentWindows = {
+		[ACTIONS.HIGH_JUMP.id]: new DriftWindow(ACTIONS.HIGH_JUMP.id, this.parser.fight.start_time),
+		[ACTIONS.GEIRSKOGUL.id]: new DriftWindow(ACTIONS.GEIRSKOGUL.id, this.parser.fight.start_time),
+	}
+
+	protected init() {
+		this.addEventHook('cast', {by: 'player', abilityId: DRIFT_ABILITIES}, this.onDriftableCast)
+		this.addEventHook('cast', {by: 'player'}, this.onCast)
+	}
+
+	private onDriftableCast(event: CastEvent) {
+		let actionId: number
+		actionId = event.ability.guid
+
+		const window = this.currentWindows[actionId]
+		window.end = event.timestamp
+		const downtime = this.downtime.getDowntime(window.start, window.end)
+		const cd = COOLDOWN_MS[actionId]
```

Comment:
nit: I know cd is a pretty common abbreviation, but could this be better as cooldown or skillCooldown instead, to be more obvious to a future maintainer?

Reply:
Fixed this in e021fe5.